compiler/rustc_trait_selection/src/traits/auto_trait.rs

```diff
@@ -344,10 +345,7 @@ impl AutoTraitFinder<'tcx> {
                 Err(SelectionError::Unimplemented) => {
                     if self.is_param_no_infer(pred.skip_binder().trait_ref.substs) {
                         already_visited.remove(&pred);
-                        self.add_user_pred(
-                            &mut user_computed_preds,
-                            pred.without_const().to_predicate(self.tcx),
-                        );
+                        self.add_user_pred(&mut user_computed_preds, pred.to_predicate(self.tcx));
```

Comment:
it's a bit subtle that `to_predicate` uses `NotConst` inside

Reply:
No it doesn't. This is converting `Binder<TraitPredicate>` to a `Predicate`. As we have a constness field in `TraitPredicate`, it does not use `NotConst` inside.